storage.lua

```diff
@@ -393,24 +393,33 @@ function mail.extractMaillists(receivers_string, maillists_owner)
 	return receivers
 end
 
-function mail.get_setting_default_value(setting_name)
-	return mail.settings[setting_name].default
+function mail.get_setting_default_value(key)
+	return mail.settings[key].default
 end
 
-function mail.get_setting(playername, setting_name)
+function mail.get_setting(playername, key)
 	local entry = mail.get_storage_entry(playername)
-	if entry.settings[setting_name] ~= nil then
-		return entry.settings[setting_name]
-	else
-		return mail.get_setting_default_value(setting_name)
+	local value = mail.get_setting_default_value(key)
+	if entry.settings[key] ~= nil then
+		value = entry.settings[key]
+	end
```

Comment:
Not really significant but could save extra call to `mail.get_setting_default_value(key)` either by this:
```suggestion
	local value = entry.settings[key] == nil
		and mail.get_setting_default_value(key)
		or entry.settings[key]
```
or similar way how it was before:
```suggestion
	local value
	if entry.settings[key] ~= nil then
		value = entry.settings[key]
	else
		value = mail.get_setting_default_value(key)
	end
```

Reply:
Btw @Athozus, I don't think it matters but should maybe have told this already in suggestion anyway:
first option (one you committed) has one special case because of how `and` / `or` actually works.
It is used like ternary operator but does not exactly work like that but instead will fall through when doing so fits in logic chain.

Basically, if there's only default setting and that default setting has falsy value then instead of returning that exact value it will return `nil` from `entry.settings` table by nonexistent `key`.

This usually works though because in Lua only falsy values are `nil` and `false`. It can however fail if you do explicit type check with `if value == false then` instead of `if not value then`.

Thought it'd be good to give heads up about this if for some reason there's such explicit checks somewhere in code.

Reply:
It is probably fine but it is also one of those things that can be a bit confusing in some edge cases.
It could be turned into real ternary by making sure that values are never falsy:
```lua
	local value = (entry.settings[key] == nil
		and {mail.get_setting_default_value(key)}
		or {entry.settings[key]})[1]
```
Trick here is to return table (always true) with single value and then use that value. It will prevent fall through when `mail.get_setting_default_value(key)` returns falsy value and expression actually works like real ternary op.

### Anyway, to actually answer your question:
I don't think any of this is needed here. It is very likely fine already, at least as long as you do not need to differentiate between `nil` and `false` for default settings.